Sources/Testing/Issues/Issue.swift

```diff
@@ -179,3 +179,225 @@ extension Issue.Kind: CustomStringConvertible {
+
+    /// An issue due to an `Error` being thrown by a test function and caught by
+    /// the testing library.
+    ///
+    /// - Parameters:
+    ///   - errorDescription: The String representation of the error which was
+    ///                       associated with this issue.
+    case errorCaught(_ errorDescription: String)
+
+    /// An issue due to a test reaching its time limit and timing out.
+    ///
+    /// - Parameters:
+    ///   - timeLimitComponents: The time limit reached by the test.
+    ///
+    /// @Comment {
+    ///   - Bug: The associated value of this enumeration case should be an
+    ///     instance of `Duration`, but the testing library's deployment target
+    ///     predates the introduction of that type.
+    /// }
+    case timeLimitExceeded(timeLimitComponents: (seconds: Int64, attoseconds: Int64))
+
+    /// A known issue was expected, but was not recorded.
+    case knownIssueNotRecorded
+
+    /// An issue occurred due to misuse of the testing library.
+    case apiMisused
+
+    /// An issue due to a failure in the underlying system, not due to a failure
+    /// within the tests being run.
+    case system
+
+    /// Snapshots an ``Issue.Kind``.
+    /// - Parameter kind: The original ``Issue.Kind`` to snapshot.
+    init(snapshotting kind: Issue.Kind) {
+      self = switch kind {
+      case .unconditional:
+          .unconditional
+      case let .expectationFailed(expectation):
+          .expectationFailed(Expectation.Snapshot(snapshotting: expectation))
+      case let .confirmationMiscounted(actual: actual, expected: expected):
+          .confirmationMiscounted(actual: actual, expected: expected)
+      case let .errorCaught(error):
+          .errorCaught(String(describing: error))
+      case let .timeLimitExceeded(timeLimitComponents: timeLimitComponents):
+          .timeLimitExceeded(timeLimitComponents: timeLimitComponents)
+      case .knownIssueNotRecorded:
+          .knownIssueNotRecorded
+      case .apiMisused:
+          .apiMisused
+      case .system:
+          .system
+      }
+    }
+
+    /// The keys used to encode ``Issue.Kind``.
+    private enum _CodingKeys: CodingKey {
+      case unconditional
+      case expectationFailed
+      case confirmationMiscounted
+      case errorCaught
+      case timeLimitExceeded
+      case knownIssueNotRecorded
+      case apiMisused
+      case system
+
+      /// The keys used to encode ``Issue.Kind.expectationFailed``.
+      enum _ExpectationFailedKeys: CodingKey {
+        case expectation
+      }
+
+      /// The keys used to encode ``Issue.Kind.confirmationMiscount``.
+      enum _ConfirmationMiscountedKeys: CodingKey {
+        case actual
+        case expected
+      }
+
+      /// The keys used to encode``Issue.Kind.errorCaught``.
+      enum _ErrorCaughtKeys: CodingKey {
+        case error
+      }
+
+      /// The keys used to encode ``Issue.Kind.timeLimitExceeds``.
+      enum _TimeLimitExceededKeys: CodingKey {
```

Comment:
Can this share code with `Clock.Instant`?

Reply:
I'll explore that in a follow-up PR.